src/main/java/embl/ebi/variation/eva/pipeline/gene/GeneLineMapper.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2016 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package embl.ebi.variation.eva.pipeline.gene;
+
+import org.springframework.batch.item.file.LineMapper;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * Created by jmmut on 2016-08-16.
+ *
+ * @author Jose Miguel Mut Lopez &lt;jmmut@ebi.ac.uk&gt;
+ */
+public class GeneLineMapper implements LineMapper<GeneMongoBean> {
+    @Override
+    public GeneMongoBean mapLine(String line, int lineNumber) throws Exception {
+        String[] lineSplit = line.split("\t");
+        String[] attributesSplit = lineSplit[8].split(";");
+        Map<String, String> attributes = new TreeMap<>();
+        for (String attribute : attributesSplit) {
+            String[] keyValue = attribute.split(" ");
+
+            // don't do a `put(keyValue[0], keyValue[1])`: a space may appear before the key
+            String value = keyValue[keyValue.length - 1];
+            String valueWithoutQuotes = value.substring(1, value.length() - 1); // remove quotes
+            attributes.put(keyValue[keyValue.length - 2], valueWithoutQuotes);
+        }
+
+        return new GeneMongoBean(attributes.get("gene_id"), attributes.get("gene_name"), lineSplit[2], lineSplit[0],
+                Integer.parseInt(lineSplit[3]), Integer.parseInt(lineSplit[4]));
```

Comment:
Looking at how GTF represents transcripts, those lines should get "transcript_id" and "transcript_name" instead of "gene_id" and "gene_name". Then the GeneMongoBean class should be renamed to something like Feature or FeatureCoordinates (it could be serialized to other databases so the reference to Mongo can be dropped).